src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -2047,7 +2047,376 @@ cdef class Polynomial(CommutativePolynomial):
+            x^7 + 20*x^6 + 16*x^5 + 36*x^4 + 59*x^3 + 3*x^2 + 46*x + 84
+            sage: g = (x^2 + 50*x + 28)
+            sage: g._any_irreducible_factor_squarefree(degree=1)
+            Traceback (most recent call last):
+            ...
+            ValueError: no irreducible factor of degree 1 could be computed from x^2 + 50*x + 28
+
+        ::
+
+            sage: # needs sage.rings.finite_rings
+            sage: F.<z4> = GF(2^4)
+            sage: R.<x> = F[]
+            sage: f = (x + z4^3 + z4^2 + z4) * (x^2 + x + z4^3 + 1) * (x^3 + (z4^3 + z4)*x^2 + z4^2 + 1)
+            sage: f._any_irreducible_factor_squarefree()
+            x + z4^3 + z4^2 + z4
+            sage: f._any_irreducible_factor_squarefree(degree=1)
+            x + z4^3 + z4^2 + z4
+            sage: f._any_irreducible_factor_squarefree(degree=2)
+            x^2 + x + z4^3 + 1
+            sage: f._any_irreducible_factor_squarefree(degree=3)
+            x^3 + (z4^3 + z4)*x^2 + z4^2 + 1
+        """
+        # If the degree is not None we only want to check a single polynomial
+        if degree is not None:
+            for (poly, d) in self._distinct_degree_factorisation_squarefree():
+                if d == degree and not poly.is_one():
+                    return poly._cantor_zassenhaus_split_to_irreducible(degree)
+                # Stop iterating early if the degree is too large
+                elif d > degree:
+                    raise ValueError(f"no irreducible factor of degree {degree} could be computed from {self}")
+            raise ValueError(f"no irreducible factor of degree {degree} could be computed from {self}")
+
+        # Otherwise we check all degrees, starting from the smallest
+        for (poly, d) in self._distinct_degree_factorisation_squarefree():
+            # Skip the split checking if `_distinct_degree_factorisation_squarefree`
+            # has found no elements of degree `d`
+            if poly.is_one():
+                continue
+
+            # Otherwise find a factor from the distinct degree factor
+            return poly._cantor_zassenhaus_split_to_irreducible(d)
+
+        raise ValueError(f"no irreducible factor could be computed from {self}")
+
+    def any_irreducible_factor(self, degree=None, assume_squarefree=False, assume_distinct_deg=False):
+        """
+        Return an irreducible factor of this polynomial.
+
+        INPUT:
+
+        - ``degree`` (None or positive integer) -- Used for polynomials
+          over finite fields. Attempts to return an irreducible factor of
+          ``self`` of chosen degree ``degree``.
+
+        - ``assume_squarefree`` (bool) -- Used for polynomials over
+          finite fields.  If ``True``, this polynomial is assumed to be
+          squarefree.
+
+        - ``assume_distinct_deg`` (bool) -- Used for polynomials over
+          finite fields.  If ``True``, this polynomial is assumed to be
+          the product of irreducible polynomials of the same degree.
```

Comment:
Here it should be mentioned that this argument only makes sense if `degree` is given as well.